src/main/java/com/hladchenko/springrestapi/controller/UserController.java

```diff
@@ -0,0 +1,60 @@
+package com.hladchenko.springrestapi.controller;
+
+import com.hladchenko.springrestapi.entity.Note;
+import com.hladchenko.springrestapi.entity.User;
+import com.hladchenko.springrestapi.service.UserService;
+import org.springframework.hateoas.EntityModel;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
+import java.util.UUID;
+
+import static org.springframework.hateoas.server.mvc.WebMvcLinkBuilder.linkTo;
+import static org.springframework.hateoas.server.mvc.WebMvcLinkBuilder.methodOn;
+
+@RequestMapping("/user")
+@RestController
+public class UserController {
+    UserService userService;
+
+    public UserController(UserService userService) {
+        this.userService = userService;
+    }
+
+    @PostMapping
+    public EntityModel<User> createUser(@RequestBody User user) throws Exception {
+        return EntityModel.of(userService.addUser(user),
+                linkTo(methodOn(UserController.class)
+                        .getUser(user.getId()))
+                        .withSelfRel());
+    }
+
+    @DeleteMapping
+    public EntityModel<User> deleteUser(@RequestParam UUID uuid) {
+        return EntityModel.of(userService.deleteUser(uuid),
+                linkTo(methodOn(UserController.class)
+                        .getUsers())
+                        .withRel("Get all users"));
+    }
+
+    @GetMapping
+    public EntityModel<User> getUser(@RequestParam UUID uuid) throws Exception {
+        return EntityModel.of(userService.getUser(uuid),
+                linkTo(methodOn(UserController.class)
+                        .deleteUser(uuid))
+                        .withRel("Delete URL"),
+                linkTo(methodOn(NoteController.class)
+                        .getNotes(uuid))
+                        .withRel("Get all user notes"));
+    }
+
+    @GetMapping("/all")
```

Comment:
for retrieving list of resources usually we do not specify /all in url. we just use plural form of resources which you want to get, e.g. GET /users